### PR TITLE
Make sure cudf uses pyarrow's libarrow during wheel builds 

### PR DESCRIPTION
**Pull request by vyasr**
## Description

This fixes an error introduced in #1304.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
CC @trxcllnt

File: ci/build_wheel_cuspatial.sh [hunk 1]
```diff
@@ -3,4 +3,6 @@
 
 set -euo pipefail
 
+export SKBUILD_CMAKE_ARGS="-DUSE_LIBARROW_FROM_PYARROW=ON"
```
Comment (trxcllnt) [inline ci/build_wheel_cuspatial.sh:6]: What if someone is building a wheel on their own? Don't they also need to pass this flag? Why can't we detect this in CMake?

Comment (vyasr) [reply ci/build_wheel_cuspatial.sh:6]: See [this comment](https://github.com/rapidsai/cudf/pull/14565#discussion_r1420589884) from @robertmaynard and [this commit](https://github.com/rapidsai/cudf/pull/14565/commits/e8a90becc91b8a380c168fa4ada5cacd781bc584) showing what I was previously doing for autodetection before removing it.

Not sure if that's the kind of detection that you're imagining.

**Review by harrism — APPROVED** [review]

**Review by trxcllnt — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Pull request merged**